sql/src/test/java/io/crate/analyze/where/WhereClauseNullReplacerTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze.where;
+
+import io.crate.analyze.symbol.Symbol;
+import io.crate.test.integration.CrateUnitTest;
+import io.crate.testing.SqlExpressions;
+import io.crate.testing.T3;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+
+public class WhereClauseNullReplacerTest extends CrateUnitTest {
+
+    private SqlExpressions sqlExpressions = new SqlExpressions(T3.SOURCES);
+
+    private void assertReplaced(String expression, String expectedString) {
+        Symbol query = sqlExpressions.asSymbol(expression);
+        Symbol replacedQuery = WhereClauseNullReplacer.replaceIfPossible(query);
+        Symbol expectedSymbol = sqlExpressions.asSymbol(expectedString);
+        assertThat(replacedQuery, is(expectedSymbol));
+    }
+
+    @Test
+    public void testNullsReplaced() throws Exception {
+        assertReplaced("null and x = null", "false and x = null");
+        assertReplaced("null or x = 1", "false or x = 1");
+        assertReplaced("not(null and x = 1)", "not(true and x = 1)");
+        assertReplaced("not(null or not(null and x = 1))", "not(true or not(false and x = 1))");
+        assertReplaced("not(null and x = 1) and not(null or x = 2)", "not(true and x = 1) and not(true or x = 2)");
+        assertReplaced("null or coalesce(null or x = 1, true)", "false or coalesce(null or x = 1, true)");
```

Comment:
Could you add some examples with other functions/operators eg: `>`, `<` to make sure that the reset of insideLogicalOperator works correctly?

Reply:
hm can do but these additional tests would cover anything in addition to the existing ones as the code doesn't care about `>` or `=`

Reply:
@matriv did you maybe read the first test over where exactly this was tested by `id = null`?

Reply:
@seut yep you're right. Was thinking that we need something that is also nested, like
`x > (y + null)` or whatever, but your expressions already test this branch of the code.